### PR TITLE
Move azure module to GA

### PR DESCRIPTION
**Pull request by narph**
Should fix some of the points in https://github.com/elastic/beats/issues/16024

- moved `azure.resource.name` to `cloud.instance.name` 
- added `cloud.instance.id` and `cloud.machine.type` (mapping the resource size for `compute_vm` and `compute_vm_scaleset`)
- added more integration tests and generated the `data.json` files
- modified dashboards based on the new fields

### CODE REVIEW

File: x-pack/metricbeat/module/azure/azure.go [hunk 1]
```diff
@@ -17,10 +17,10 @@ import (
 
 // Config options
 type Config struct {
-	ClientID            string           `config:"client_id"    validate:"required"`
+	ClientId            string           `config:"client_id"    validate:"required"`
 	ClientSecret        string           `config:"client_secret" validate:"required"`
-	TenantID            string           `config:"tenant_id" validate:"required"`
-	SubscriptionID      string           `config:"subscription_id" validate:"required"`
+	TenantId            string           `config:"tenant_id" validate:"required"`
+	SubscriptionId      string           `config:"subscription_id" validate:"required"`
```
Comment (exekias) [inline x-pack/metricbeat/module/azure/azure.go]: Why these changes? I don't think there is anything wrong with ID

File: x-pack/metricbeat/module/azure/container_instance/container_instance_integration_test.go [hunk 2]
```diff
@@ -2,50 +2,41 @@
...
-	}
-	config := map[string]interface{}{
-		"module":          "azure",
-		"metricsets":      []string{"container_instance"},
-		"client_id":       clientId,
-		"client_secret":   clientSecret,
-		"tenant_id":       tenantId,
-		"subscription_id": subscriptionId,
+func TestData(t *testing.T) {
+	config, err := test.GetConfig("container_instance")
```
Comment (exekias) [inline x-pack/metricbeat/module/azure/container_instance/container_instance_integration_test.go]: :clap:

File: x-pack/metricbeat/module/azure/azure.go (hunk 1 above)
Comment (narph) [reply x-pack/metricbeat/module/azure/azure.go]: there was a mix of `Id` and `ID` in the code so I went with most used for consistency

**Comment by narph** [comment]
cc @sorantis, just a heads up on the changes, dashboards have been updated as well
